src/catalog/src/durable/impls/persist.rs

```diff
@@ -127,11 +131,36 @@ impl PersistHandle {
             )));
         }
 
-        let user_version = self.get_user_version(upper).await;
+        // Listen cannot read any writes that happen at `Timestamp::minimum()`. So we initialize
```

Comment:
nit: we're still technically using a separate Snapshot+Listen instead of a Subscribe, but every time I read this comment the reference to using Listens directly causes me to pause. I think we can resolve it by being slightly more vague: something like first sentence with  "It's convenient to establish the invariant that the shard is always readable at _some_ timestamp, even if it's not yet initialized. ..." and leave the rest as-is

Reply:
Updated